pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -231,10 +231,12 @@ func (s *DiscoveryServer) periodicRefreshMetrics() {
 	defer ticker.Stop()
 	for range ticker.C {
 		push := s.globalPushContext()
+		push.Mutex.Lock()
```

Comment:
It looks like more than fixing a flake here. We're fixing a bug. Do we know when this bug was introduced and why it wasn't caught by our tests?

cc @costinm @rshriram

Reply:
Hi @nmittler ,

Could this pr be approved to fix the bug in release-1.0? Or could I do something further?

Thanks.

Reply:
This race fix has been fixed in master/1.1 as part of #10036

Reply:
@duderino what is the policy for commit to the release-1.0 branch? Are we still releasing 1.0 patches?

Reply:
@nmittler I found #11616 was merged to release-1.0 branch last week.